controllers/eventhandlers/targetgrouppolicy.go

```diff
@@ -0,0 +1,158 @@
+		h.log.Infow("Detected cross namespace TargetGroupPolicy attachment, skipping",
+			"policyName", policyName, "targetRef", targetRef)
+		return nil
+	}
+
+	svcName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      string(targetRef.Name),
+	}
+	svc := &corev1.Service{}
+	if err := h.client.Get(context.TODO(), svcName, svc); err != nil {
+		if errors.IsNotFound(err) {
+			h.log.Debugw("TargetGroupPolicy is referring to non-existent service, skipping",
+				"policyName", policyName, "serviceName", svcName.String())
+		} else {
+			// Still gracefully skipping the event but errors other than NotFound are bad sign.
+			h.log.Warnw("Failed to query targetRef of TargetGroupPolicy",
+				"policyName", policyName, "serviceName", svcName.String(), "reason", err.Error())
+		}
+		return nil
+	}
+	h.log.Debugw("TargetGroupPolicy change on Service detected",
+		"policyName", policyName, "serviceName", svcName.String())
+
+	return svc
+}
+
+func (h *targetGroupPolicyEventHandler) MapToHTTPRoute(obj client.Object) []reconcile.Request {
+	var requests []reconcile.Request
+
+	svc := h.getTargetRef(obj)
+	if svc == nil {
+		return nil
+	}
+
+	routeList := &gateway_api.HTTPRouteList{}
+	h.client.List(context.TODO(), routeList)
+
+	for _, httpRoute := range routeList.Items {
+		if isServiceUsedByRoute(core.NewHTTPRoute(httpRoute), svc) {
+			routeName := types.NamespacedName{
+				Namespace: httpRoute.Namespace,
+				Name:      httpRoute.Name,
+			}
+			requests = append(requests, reconcile.Request{NamespacedName: routeName})
+			h.log.Infow("Service TargetGroupPolicy change triggering HTTPRoute update",
+				"serviceName", svc.Namespace+"/"+svc.Name, "routeName", routeName.String())
+		}
+	}
+	return requests
+}
+
+func (h *targetGroupPolicyEventHandler) MapToGRPCRoute(obj client.Object) []reconcile.Request {
+	var requests []reconcile.Request
+
+	svc := h.getTargetRef(obj)
+	if svc == nil {
+		return nil
+	}
+
+	routeList := &gateway_api_v1alpha2.GRPCRouteList{}
+	h.client.List(context.TODO(), routeList)
+
+	for _, grpcRoute := range routeList.Items {
+		if isServiceUsedByRoute(core.NewGRPCRoute(grpcRoute), svc) {
+			routeName := types.NamespacedName{
+				Namespace: grpcRoute.Namespace,
+				Name:      grpcRoute.Name,
+			}
+			requests = append(requests, reconcile.Request{NamespacedName: routeName})
+			h.log.Infow("Service TargetGroupPolicy change triggering GRPCRoute update",
+				"serviceName", svc.Namespace+"/"+svc.Name, "routeName", routeName.String())
+		}
+	}
+	return requests
+}
+
+func (h *targetGroupPolicyEventHandler) MapToServiceExport(obj client.Object) []reconcile.Request {
+	var requests []reconcile.Request
+
+	svc := h.getTargetRef(obj)
+	if svc == nil {
+		return nil
+	}
+	svcName := types.NamespacedName{
+		Namespace: svc.Namespace,
+		Name:      svc.Name,
+	}
+	svcExport := &mcs_api.ServiceExport{}
+	if err := h.client.Get(context.TODO(), svcName, svcExport); err != nil {
+		return nil
```

Comment:
In this case we would expect the ```.Get``` to succeed, right? is it worth logging here or is this actually expected?

Reply:
yep it is worth logging here. will fix